packages/qwik/src/testing/library.ts

```diff
@@ -0,0 +1,37 @@
+import type { JSXNode } from '@builder.io/qwik/jsx-runtime';
+import { render as renderIn } from '@builder.io/qwik';
+import { ElementFixture, trigger } from './element-fixture';
+import { setTestPlatform } from './platform';
+
+/**
+ * @private
+ * @param element
+ * @returns
+ */
+const parseClassWithElement = (element: Element) => {
+  if (!element.classList.value.length) return [];
+  return element.classList.value
+    .split(' ')
+    .map((c) => `.${c}`)
+    .join('');
+};
+
+/**
+ * CreatePlatfrom and CreateDocument
+ * @alpha
+ */
+export const createDOM = function () {
+  setTestPlatform();
+  const host = new ElementFixture().host;
+  return {
+    render: function (jsxElement: JSXNode) {
+      return renderIn(host, jsxElement);
+    },
+    screen: host,
+    userEvent: function (element: HTMLElement | string | Element, event: string) {
+      if (typeof element === 'string') return trigger(host, element, event);
+      const classListValue = parseClassWithElement(element);
```

Comment:
i am a bit confused by this

Reply:
> I think we should modified `trigger` function to receive both CSS selector and element instead of.

But `trigger` already has this function. 

```tsx
// card
 <button class='block'>Hello!</button >
```


```tsx 
// test
 await userEvent('.block', 'click');
```

Reply:
I mean
```ts
export async function trigger(
  root: Element,
  selector: string,
  eventNameCamel: string
): Promise<void> {
  for (const element of Array.from(root.querySelectorAll(selector))) {
```
to

```ts
export async function trigger(
  root: Element,
  selector: string | Element, // ArrayLike<Element> | Iterable<Element>
  eventNameCamel: string
): Promise<void> {
const elements = typeof selector === 'string' ? Array.from(root.querySelectorAll(selector)) : [selector]
  for (const element of elements) {
```

So we can call `trigger` directly without weird part like `parseClassWithElement`

Btw, `parseClassWithElement` fail with this case below:
```tsx
// buttons
<button>Button 1</button>
<button>Button 2</button>
```
```ts
// test
const button = root.querySelector("button:nth-child(2)")
await userEvent(button, 'click'); // selector = 'BUTTON'
```

P/s: Sorry, I'm not a native English speaker

Reply:
I get it the point. Yeah I think that is possible without complication. Let me try. Also, I'm working to move some functions

Reply:
@manucorporat Do you know or have any idea about it?

I have two buttons but `document.querySelector` only work when I use `document.querySelector('button:nth-child(4)')`  

![image](https://user-images.githubusercontent.com/15802366/198060999-8d05bb37-dcdd-4875-87c7-a2635e17ea6f.png)

Reply:
but the way @langbamit thanks for your suggestion. Added 🙌